basalt_runner/src/dataset_io_jsonl.h

```diff
@@ -184,7 +189,7 @@ class JsonlVioDataset : public VioDataset {
 
         size_t full_size = img.cols * img.rows;
         for (size_t i = 0; i < full_size; i++) {
-          int val = data_in[i * 3];
+          int val = (data_in[i * 3] + data_in[i * 3 + 1] + data_in[i * 3 + 2]) / 3;
```

Comment:
It probably doesn't matter much, but this doesn't produce completely identical results with cvtColor(), because it uses 0.299⋅R+0.587⋅G+0.114⋅B formula.

https://docs.opencv.org/3.4/de/d25/imgproc_color_conversions.html#color_convert_rgb_lab

Reply:
Yes, my intention was to just roughly fix the original line which used the blue channel, as that could be bad if the video was actually in color.